cluster/gce/upgrade.sh

```diff
@@ -241,24 +241,31 @@ function prepare-node-upgrade() {
   echo "== Preparing node upgrade (to ${KUBE_VERSION}). ==" >&2
   setup-base-image
 
-  SANITIZED_VERSION=$(echo ${KUBE_VERSION} | sed 's/[\.\+]/-/g')
+  SANITIZED_VERSION="${KUBE_VERSION//[\.\+]/-}"
 
   # TODO(zmerlynn): Refactor setting scope flags.
   local scope_flags=
   if [ -n "${NODE_SCOPES}" ]; then
     scope_flags="--scopes ${NODE_SCOPES}"
   else
+    # shellcheck disable=SC2034 # 'scope_flags' is used by upstream
     scope_flags="--no-scopes"
   fi
 
   # Get required node env vars from exiting template.
-  local node_env=$(get-node-env)
+  local node_env
+  node_env=$(get-node-env)
   KUBE_PROXY_TOKEN=$(get-env-val "${node_env}" "KUBE_PROXY_TOKEN")
+  export KUBE_PROXY_TOKEN
   NODE_PROBLEM_DETECTOR_TOKEN=$(get-env-val "${node_env}" "NODE_PROBLEM_DETECTOR_TOKEN")
   CA_CERT_BASE64=$(get-env-val "${node_env}" "CA_CERT")
+  export CA_CERT_BASE64
   EXTRA_DOCKER_OPTS=$(get-env-val "${node_env}" "EXTRA_DOCKER_OPTS")
+  export EXTRA_DOCKER_OPTS
   KUBELET_CERT_BASE64=$(get-env-val "${node_env}" "KUBELET_CERT")
+  export KUBELET_CERT_BASE64
   KUBELET_KEY_BASE64=$(get-env-val "${node_env}" "KUBELET_KEY")
+  export KUBELET_KEY_BASE64
```

Comment:
why are these exported? what needs them?

Reply:
According to https://github.com/kubernetes/kubernetes/pull/7938#discussion_r29965211, these variables are required by code around https://github.com/kubernetes/kubernetes/blob/0926c9c47677e944c2dadcd7b39587866331eeba/cluster/gce/util.sh#L1182